### PR TITLE
BUG: mixedlm fit_regularized, missing vcomp in results

### LINKED ISSUE #8681: Cannot access random_effects parameter after training mixedlm with regularization.
#### Describe the bug

Cannot access random_effects parameter after training mixedlm with regularization (```fit_regularized()```).

After successfully fitting LME model with ```fit_regularized```, when I try to access random_effects from ```MixedLMResults``` object, I'll get error: 

```AttributeError: 'MixedLMResults' object has no attribute 'vcomp'```

image

This does not happen when fitting LME model with ```fit``` function. 
It is caused by missing attribute 'vcomp', that is not assigned in ```fit_regularized()``` function, when creating results. 

Also, when trying to call summary function on MixedLMResults object that was fitted with fit_regularized, it fails with this error:

image

#### Code Sample, a copy-pastable example if possible

```python
import numpy as np
import pandas as pd
import statsmodels.formula.api as smf


data = {'target': np.random.randint(2, size=10),
     'feature_1': np.random.randint(2, size=10),
     'feature_2': np.random.randint(2, size=10),
     'group': ['a','a','a','a','a','b','b','b','b','b']}
df = pd.DataFrame(data=data)
md = smf.mixedlm(formula=f"target ~ feature_1 + feature_2",
                data=df, groups=df["group"],
                re_formula=f"~ feature_1")

results = md.fit_regularized()
results.random_effects
```

#### Expected Output

Should return dictionary with random_effects.

#### Output of ``import statsmodels.api as sm; sm.show_versions()``

<details>

[paste the output of ``import statsmodels.api as sm; sm.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
Python: 3.10.5.final.0

statsmodels
===========

Installed: 0.13.5 (c:\Users\Juraj_Mecir\.pyenv\pyenv-win\versions\3.10.5\lib\site-packages\statsmodels)

Required Dependencies
=====================

cython: Not installed
numpy: 1.22.3 (c:\Users\Juraj_Mecir\.pyenv\pyenv-win\versions\3.10.5\lib\site-packages\numpy)
scipy: 1.7.3 (c:\Users\Juraj_Mecir\.pyenv\pyenv-win\versions\3.10.5\lib\site-packages\scipy)
pandas: 1.5.1 (c:\Users\Juraj_Mecir\.pyenv\pyenv-win\versions\3.10.5\lib\site-packages\pandas)
    dateutil: 2.8.2 (C:\Users\Juraj_Mecir\AppData\Roaming\Python\Python310\site-packages\dateutil)
patsy: 0.5.3 (c:\Users\Juraj_Mecir\.pyenv\pyenv-win\versions\3.10.5\lib\site-packages\patsy)

Optional Dependencies
=====================

matplotlib: 3.6.0 (c:\Users\Juraj_Mecir\.pyenv\pyenv-win\versions\3.10.5\lib\site-packages\matplotlib)
    backend: module://matplotlib_inline.backend_inline 
cvxopt: Not installed
joblib: 1.2.0 (c:\Users\Juraj_Mecir\.pyenv\pyenv-win\versions\3.10.5\lib\site-packages\joblib)

Developer Tools
================

IPython: 8.7.0 (C:\Users\Juraj_Mecir\AppData\Roaming\Python\Python310\site-packages\IPython)
    jinja2: 3.1.2 (c:\Users\Juraj_Mecir\.pyenv\pyenv-win\versions\3.10.5\lib\site-packages\jinja2)
sphinx: Not installed
    pygments: 2.13.0 (C:\Users\Juraj_Mecir\AppData\Roaming\Python\Python310\site-packages\pygments)
pytest: Not installed
virtualenv: Not installed

</details>

### PR DESCRIPTION
**Pull request by JuroMecir**
- [x] closes #8681 
- [x] tests added / passed. Added test for summary when training LME with train_regularized.

### CODE REVIEW

File: statsmodels/regression/tests/test_lme.py [hunk 1]
```diff
@@ -803,6 +803,51 @@ def test_summary_xname_re(self):
 # ------------------------------------------------------------------
 
 
+class TestMixedLMSummaryRegularized:
```
Comment (josef-pkt) [inline statsmodels/regression/tests/test_lme.py:806]: are the test methods here exact copies of the unregularized version?

If yes, then you can just subclass TestMixedLMSummary and only define the `setup_class` methods, inheriting the test methods

Comment (JuroMecir) [reply statsmodels/regression/tests/test_lme.py:806]: Will do.

Comment (JuroMecir) [reply statsmodels/regression/tests/test_lme.py]: Done

**Comment by josef-pkt** [comment]
Thanks for the bug report and pull request.

LGTM, merged

**Pull request merged**